packages/OmegaPrint-Tests.package/TestUI.class/instance/testFormatStringWithButton.st

```diff
@@ -0,0 +1,22 @@
+button functionality
+testFormatStringWithButton
+	
+	| browser cst evaluator |
+	cst := OhmExplicitSendsSmalltalk 
+				match: (self class
+								ultimateSourceCodeAt: #exampleMethod
+								ifAbsent: [ ^ 'error' ])
+				startingFrom: #MethodDeclaration.
+	evaluator := OhmExplicitSendsSmalltalk synthesizedAttribute: OPPrinter new.
+	browser := Browser new.
+	browser contentsSymbol: #omegaPrint.
+	browser
+		instVarNamed: 'selectedClassName'
+		put: #TestUI.
+	browser
+		instVarNamed: 'selectedMessageName'
+		put: #exampleMethod.
+
+	self
+		assert: browser sourceStringPrettifiedAndDiffed
+		equals: (evaluator value: cst)
```

Comment:
Auch hier haette ich das erwartete Ergebnis lieber als String. Wenn unser `OPPrinter` einen Fehler hätte, koennte der ja durchaus in `browser sourceStringPrettifiedAndDiffed` und `evaluator value: cst` auftreten und sich damit aufheben.

Reply:
das ist so gewollt. Wir wollen hier ja nur testen, ob der Button geht und nicht ob das richtig formatiert wird.

Reply:
wenn wir das mit einem String vergleichen, dann müssen wir mit jeder neuen Regel, die wir implementieren auch den String anpassen, obwohl wir die Buttonfunktionaltiät prüfen wollen

Reply:
klingt sinnvoll